From 2fcf4605eded954159a61e9f298294730d7217be Mon Sep 17 00:00:00 2001
From: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Date: Tue, 31 May 2016 16:59:10 -0300
Subject: [PATCH 58/88] staging/android: remove sync_timeline_destroy()
Git-commit: 711102325302d793aa0eee493be79ce9c32ae426
Patch-mainline: v4.8-rc1
References: fate#321223

This function was just used by the file release function, so we just fold
its content there and remove sync_timeline_destroy().

Signed-off-by: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Reviewed-by: Sumit Semwal <sumit.semwal@linaro.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/staging/android/sw_sync.c | 19 +++----------------
 1 file changed, 3 insertions(+), 16 deletions(-)

diff --git a/drivers/staging/android/sw_sync.c b/drivers/staging/android/sw_sync.c
index 1f956b99b880..cf4de27266a4 100644
--- a/drivers/staging/android/sw_sync.c
+++ b/drivers/staging/android/sw_sync.c
@@ -98,21 +98,6 @@ static void sync_timeline_put(struct sync_timeline *obj)
 }
 
 /**
- * sync_timeline_destroy() - destroys a sync object
- * @obj:	sync_timeline to destroy
- *
- * A sync implementation should call this when the @obj is going away
- * (i.e. module unload.)  @obj won't actually be freed until all its children
- * fences are freed.
- */
-static void sync_timeline_destroy(struct sync_timeline *obj)
-{
-	smp_wmb();
-
-	sync_timeline_put(obj);
-}
-
-/**
  * sync_timeline_signal() - signal a status change on a sync_timeline
  * @obj:	sync_timeline to signal
  * @inc:	num to increment on timeline->value
@@ -275,7 +260,9 @@ static int sw_sync_debugfs_release(struct inode *inode, struct file *file)
 {
 	struct sync_timeline *obj = file->private_data;
 
-	sync_timeline_destroy(obj);
+	smp_wmb();
+
+	sync_timeline_put(obj);
 	return 0;
 }
 
-- 
2.12.0
